src/libgit2/fetch.c

```diff
@@ -60,7 +61,7 @@ static int mark_local(git_remote *remote)
 
 	git_vector_foreach(&remote->refs, i, head) {
 		/* If we have the object, mark it so we don't ask for it */
-		if (git_odb_exists(odb, &head->oid))
+		if (remote->nego.depth != INT_MAX && git_odb_exists(odb, &head->oid))
```

Comment:
Could you put a comment explaining why this is needed, just for future reference.